src/ContainerManagement.js

```diff
@@ -1,35 +1,60 @@
 const _forEach = require('lodash.foreach');
+const _get = require('lodash.get');
+const _isFunction = require('lodash.isfunction');
+const _isObject = require('lodash.isobject');
 const Dependency = require('./Dependency');
 
 const rall = /.+/;
 
 module.exports = {
 
-  warnIfNeeded(name) {
+  getDependencySourceHint(dependency) {
+    if (_isFunction(dependency)) {
+      return _get(dependency, 'name') || '<anonymous function>';
+    }
+
+    if (_isObject(dependency)) {
+      return _get(dependency, 'constructor.name') || '<object>';
+    }
+
+    return '<primitive>';
+  },
+
+  warnIfNeeded(name, dependency) {
     if (this.hasDependency(name)) {
-      this.logger.warn(`warning: ${name} dependency is being overwritten in ${this.name} injector`);
+      const hint = this.getDependencySourceHint(dependency);
+      this.logger.warn(`warning: ${name} (${hint}) dependency is being overwritten in ${this.name} injector`);
     }
   },
 
   addResolvableDependency(name, dependency, suppressWarning = false) {
+    if (this.shouldIgnoreDependency(dependency)) {
+      return this;
+    }
     if (!suppressWarning) {
-      this.warnIfNeeded(name);
+      this.warnIfNeeded(name, dependency);
     }
     this.dependencies[name] = Dependency.resolvableDependency(name, dependency);
     return this;
   },
 
   addDependency(name, dependency, suppressWarning = false) {
+    if (this.shouldIgnoreDependency(dependency)) {
+      return this;
```

Comment:
Wondering if we should add a debug message here that says we are ignoring this dependency..

Reply:
@acolchado updated to print a warning if `suppressWarning` is false.